oauth2client/client.py

```diff
@@ -1843,6 +1843,22 @@ def __init__(self, client_id,
         }
         self.params.update(kwargs)
 
+        self._check_for_approval_prompt()
```

Comment:
Cool, so just pull that out into a module-level function? Sounds good.

I did this originally because I didn't feel like this logic belonged in the constructor, so I'm fine with pulling it into a separate function.